inductiva/resources/machines_base.py

```diff
@@ -60,19 +73,20 @@ def _register_machine_group(self, **kwargs):
         self._id = resp.body["id"]
         self._name = resp.body["name"]
         self.register = False
+        self._log_machine_group_info()
 
     @classmethod
     def from_api_response(cls, resp: dict):
         """Creates a MachineGroup object from an API response."""
 
         machine_group = cls(
             machine_type=resp["machine_type"],
-            spot=bool(resp["spot"]),
             disk_size_gb=resp["disk_size_gb"],
             register=False,
         )
         machine_group._id = resp["id"]
         machine_group._name = resp["name"]
+        machine_group.spot = resp["spot"]
```

Comment:
I think this attribute only makes sense to exist in a Derived class that actually supports spot.
spot is non-sensical in the context of the MPICluster and does not make sense to have this attr available there